### PR TITLE
Add input/output bindings tests

### PR DESCRIPTION
**Pull request by philliphoff**
# Description

Adds new tests related to input and output bindings.  Some refactoring of existing tests to make use of new, common infrastructure.

> Contains two skipped tests that uncovered #28 that will be enabled as part of a fix on a separate PR.

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: src/Dapr.PluggableComponents.Tests/Adaptors/InputBindingAdaptorTests.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    [Fact]
+    public Task PingTests()
+    {
+        return AdaptorFixture.TestPingAsync<InputBindingAdaptor, IInputBinding>(
+            AdaptorFixture.CreateInputBinding,
+            fixture => fixture.Adaptor.Ping(new PingRequest(), fixture.Context));
+    }
+
+    [Fact(Timeout = TimeoutInMs, Skip = "To be re-enabled when as part of fix for dapr-sandbox/components-dotnet-sdk#28.")]
```
Comment (tmacam) [inline src/Dapr.PluggableComponents.Tests/Adaptors/InputBindingAdaptorTests.cs:44]: 👍

File: src/Dapr.PluggableComponents.Tests/Adaptors/InputBindingAdaptorTests.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            fixture.Context);
+
+        await pullMessagesTask;
+
+        fixture.MockComponent
+            .Verify(
+                component => component.ReadAsync(
+                    It.IsAny<MessageDeliveryHandler<InputBindingReadRequest, InputBindingReadResponse>>(),
+                    It.Is<CancellationToken>(token => token == fixture.Context.CancellationToken)),
+                Times.Once());
```
Comment (tmacam) [inline src/Dapr.PluggableComponents.Tests/Adaptors/InputBindingAdaptorTests.cs:125]: The tests verify `ReadAsync` was called but but it isn't verifying a message was handled, or is it?

File: src/Dapr.PluggableComponents.Tests/Adaptors/OutputBindingAdaptorTests.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        fixture.MockComponent
+            .Setup(component => component.ListOperationsAsync(It.Is<CancellationToken>(token => token == fixture.Context.CancellationToken)))
+            .ReturnsAsync(operations);
+
+        var response = await fixture.Adaptor.ListOperations(new ListOperationsRequest(), fixture.Context);
+
+        Assert.Equal(operations, response.Operations);
+
+        fixture.MockComponent
+            .Verify(component => component.ListOperationsAsync(It.Is<CancellationToken>(token => token == fixture.Context.CancellationToken)), Times.Once);
```
Comment (tmacam) [inline src/Dapr.PluggableComponents.Tests/Adaptors/OutputBindingAdaptorTests.cs:78]: We should verify the number of operations returned.

File: src/Dapr.PluggableComponents.Tests/Adaptors/OutputBindingAdaptorTests.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+namespace Dapr.PluggableComponents.Adaptors;
+
+public sealed class OutputBindignAdaptorTests
+{
+    [Fact]
+    public Task InitTests()
+    {
+        return AdaptorFixture.TestInitAsync(
+            () => AdaptorFixture.CreateOutputBinding(),
+            (fixture, metadataRequest) => fixture.Adaptor.Init(new Proto.Components.V1.OutputBindingInitRequest { Metadata = metadataRequest }, fixture.Context));
```
Comment (tmacam) [inline src/Dapr.PluggableComponents.Tests/Adaptors/OutputBindingAdaptorTests.cs:29]: nice.

**Review by tmacam — APPROVED** [review]
Overall LGTM. If I would suggest something would be to not only verify that the calls were made but that the data received matches the expectation.

File: src/Dapr.PluggableComponents.Tests/Adaptors/InputBindingAdaptorTests.cs (hunk 2 above)
Comment (philliphoff) [reply src/Dapr.PluggableComponents.Tests/Adaptors/InputBindingAdaptorTests.cs:125]: It does, but in an indirect way, in that the test won't end (and therefore timeout) if no message was actually delivered or ack'd.

File: src/Dapr.PluggableComponents.Tests/Adaptors/OutputBindingAdaptorTests.cs (hunk 3 above)
Comment (philliphoff) [reply src/Dapr.PluggableComponents.Tests/Adaptors/OutputBindingAdaptorTests.cs:78]: The `Assert.Equal(operations, response.Operations);` above does that (as `Assert.Equal()` ensures collections are equivalent).

**Pull request merged**